nixos/modules/services/networking/v2raya.nix

```
@@ -6,11 +6,22 @@ with lib;
   options = {
     services.v2raya = {
       enable = options.mkEnableOption "the v2rayA service";
+
+      cliPackage = mkOption {
+        type = types.package;
+        default = pkgs.v2ray;
+        defaultText = literalExpression "pkgs.v2ray";
+        example = literalExpression "pkgs.xray";
+        description = ''
+          The package used for the cli binary.
+        '';
+        relatedPackages = [ "xray" "v2ray" ];
+      };
     };
   };
 
   config = mkIf config.services.v2raya.enable {
-    environment.systemPackages = [ pkgs.v2raya ];
+    environment.systemPackages = [ (pkgs.v2raya.override { v2ray = config.services.v2raya.cliPackage; }) ];
```

This seems unnecessary complex and inflexible. 

```suggestion
    environment.systemPackages = [ cfg.package ];
```

This makes the configuration more straight forward and flexible, and people can also just overwrite the cli package or the entire package